compiler/runtime/OMRRelocationRecord.cpp

```diff
@@ -0,0 +1,393 @@
+   {
+   TR::RelocationRecordBinaryTemplate *recordPointer = firstRecord(reloTarget);
+   TR::RelocationRecordBinaryTemplate *endOfRecords = pastLastRecord(reloTarget);
+
+   while (recordPointer < endOfRecords)
+      {
+      TR::RelocationRecord storage;
+      // Create a specific type of relocation record based on the information
+      // in the binary record pointed to by `recordPointer`
+      TR::RelocationRecord *reloRecord = TR::RelocationRecord::create(&storage, reloRuntime, reloTarget, recordPointer);
+      int32_t rc = handleRelocation(reloRuntime, reloTarget, reloRecord, reloOrigin);
+      if (rc != 0)
+         return rc;
+
+      recordPointer =  reinterpret_cast<TR::RelocationRecordBinaryTemplate *>(reloRecord->nextBinaryRecord(reloTarget));
+      }
+
+   return 0;
+   }
+void OMR::RelocationRecord::initialize(TR::RelocationTarget *reloTarget,uint8_t* targetAddress, uint8_t* targetAddress2 ){
+
+}
+
+int32_t
+OMR::RelocationRecordGroup::handleRelocation(TR::RelocationRuntime *reloRuntime,
+                                           TR::RelocationTarget *reloTarget,
+                                           TR::RelocationRecord *reloRecord,
+                                           uint8_t *reloOrigin)
+   {
+
+   if (reloRecord->ignore(reloRuntime))
+      {
+      return 0;
+      }
+
+   reloRecord->preparePrivateData(reloRuntime, reloTarget);
+   return reloRecord->applyRelocationAtAllOffsets(reloRuntime, reloTarget, reloOrigin);
+   }
+
+#define FLAGS_RELOCATION_WIDE_OFFSETS   0x80
+#define FLAGS_RELOCATION_EIP_OFFSET     0x40
+#define FLAGS_RELOCATION_TYPE_MASK      (TR::ExternalRelocationTargetKindMask)
+#define FLAGS_RELOCATION_FLAG_MASK      ((uint8_t) (FLAGS_RELOCATION_WIDE_OFFSETS | FLAGS_RELOCATION_EIP_OFFSET))
+
+
+TR::RelocationRecord *
+TR::RelocationRecord::create(TR::RelocationRecord *storage, TR::RelocationRuntime *reloRuntime, TR::RelocationTarget *reloTarget, TR::RelocationRecordBinaryTemplate *record)
+   {
+   OMR::RelocationRecord *reloRecord = NULL;
+   // based on the type of the relocation record, create an object of a particular variety of OMR::RelocationRecord object
+   uint8_t reloType = record->type(reloTarget);
+   switch (reloType)
+      {
+      default:
+         // TODO: error condition
+         printf("Unexpected relo record: %d\n", reloType);fflush(stdout);
+         exit(0);
+      }
+   
+   return static_cast<TR::RelocationRecord*>(reloRecord);
+   }
+
+void
+OMR::RelocationRecord::print(TR::RelocationRuntime *reloRuntime)
+   {
+   TR::RelocationTarget *reloTarget = reloRuntime->reloTarget();
+   }
+
+void
+OMR::RelocationRecord::clean(TR::RelocationTarget *reloTarget)
+   {
+   self()->setSize(reloTarget, 0);
+   reloTarget->storeUnsigned8b(0, (uint8_t *) &_record->_type);
+   reloTarget->storeUnsigned8b(0, (uint8_t *) &_record->_flags);
+   }
+
+int32_t
+OMR::RelocationRecord::bytesInHeader()
+   {
+   return sizeof(OMR::RelocationRecordBinaryTemplate);
```

Comment:
This should probably just be `return OMR::RelocationRecord::getSizeOfAOTRelocationHeader(self()->type());`

Reply:
Hello, thank you for the review! I will try to resolve the problem and ship the solution with the next commit